library/Icingadb/Widget/Detail/QuickActions.php

```diff
@@ -104,7 +107,10 @@ protected function assemble()
             );
         }
 
-        if ($this->isGrantedOn('icingadb/command/process-check-result', $this->object)) {
+        if (
+            $this->isGrantedOn('icingadb/command/process-check-result', $this->object)
+            && $this->object->passive_checks_enabled
```

Comment:
Check the enable flag first, it increases efficiency as it's less costly to check